src/main/java/com/example/ticketbookingapp/controllers/MovieController.java

```diff
@@ -0,0 +1,66 @@
+package com.example.ticketbookingapp.controllers;
+
+import com.example.ticketbookingapp.models.Movie;
+import com.example.ticketbookingapp.services.interfaces.IMovieService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@RestController
+@RequestMapping("api/movies")
+public class MovieController {
+    private final IMovieService service;
+
+    public MovieController(IMovieService service) {
+        this.service = service;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<Movie>> getAllMovies() {
+        return ResponseEntity.ok(service.getAllMovies());
+    }
+
+    @GetMapping("{id}")
+    public ResponseEntity getMovie(@PathVariable Long id) {
+        try {
+            return ResponseEntity.ok(service.getMovie(id));
+        } catch (NoSuchElementException exc) {
```

Comment:
O kurczaki, co tu się dzieje?

Reply:
Rozumiem, że to nie jest zbyt dobra praktyka, żeby przechwytywać w ten sposób wyjątki w kontrolerze? :(

Reply:
Zdecydowanie źle jest przechwytywać (i rzucać!) wyjątki ogólnojavowe. Po pierwsze - lepiej zrobić własne wyjątki (zwróć uwagę, że ten wyjątek jest bardzo ogólny, mógł polecieć z miejsca z którego wcale się nie spodziewasz). Po drugie - popatrz na controller advice.